saleor/checkout/complete_checkout.py

```diff
@@ -1516,7 +1547,10 @@ def complete_checkout_with_payment(
         lines, _ = fetch_checkout_lines(checkout, skip_recalculation=True)
         checkout_info = fetch_checkout_info(checkout, lines, manager)
 
-        checkout_info.checkout.voucher_code = None
+        # reassign voucher data that was used during payment process to allow voucher
+        # usage releasing in case of checkout complete failure
+        checkout_info.voucher = voucher
```

Comment:
Shouldn't be passed to `fetch_checkout_info(`? There is an input `voucher` there 🤔 If not added it makes additional calls to DB to fetch the voucher and code